include/wasmtime.hh

```diff
@@ -2454,12 +2472,14 @@ public:
    * > signature is statically known it's recommended to use `Func::typed` and
    * > `TypedFunc::call`.
    */
+  template<typename I>
   TrapResult<std::vector<Val>> call(Store::Context cx,
-                                    const std::vector<Val> &params) const {
+                                    const I &begin,
+                                    const I &end) const {
     std::vector<wasmtime_val_t> raw_params;
-    raw_params.reserve(params.size());
-    for (const auto &param : params) {
-      raw_params.push_back(param.val);
+    raw_params.reserve(end - begin);
+    for (auto i = begin; i != end; i++) {
```

Comment:
also kind of a super nit, but with C++ iterators you generally want to pre-increment, as post increment will make a copy and that copy constructor might not be optimized away by the compiler.

Resources: http://www.gotw.ca/gotw/002.htm, https://stackoverflow.com/questions/1077026/incrementing-iterators-is-it-more-efficient-than-it